### ISSUE TITLE
Predictable link-following behaviour

### ISSUE THREAD

**Issue by daniel-koudouna** [jethrokuan/org-roam#30]
I'm wondering if anyone else is having inconsistent behaviour when following links.

The only way to follow links from org seems to be `org-open-at-point`. This doesn't mirror the behaviour of `find-file` for me. It sometimes opens a new window, and the org-roam buffer is updated with a considerable delay. For evil users, this also messes with evil jumping.

I've written a find function to use more predictable behaviour, Would this be useful to add as a PR?  As I don't use vanilla org-mode features, this might be a solved problem.

**Comment by jethrokuan** [comment]
This is right, the org-roam buffer won't update on an `org-open-file-at-point`. That's either because the `post-command-hook` doesn't get run, or when it was run it the predicates don't pass. It gets updated on the next command.

There might be a more proper solution here, like advicing the function. The "opening in a new window" behaviour is also probably customizable somehow. I'll look at it later.

**Comment by jethrokuan** [comment]
Regarding `org-open-at-point`: This helps alleviate the problem a little:

`(add-to-list 'org-link-frame-setup '(file . find-file))`

It helps when opening file links within the main org buffer, but it doesn't help for opening links in the org-roam buffer, because that's a side window. Seems like `org-open-at-point` eventually runs the `post-command-hook` and updates the org-roam-buffer though.

**Comment by daniel-koudouna** [comment]
Would it be worthwhile to have a custom link following function for the org-roam buffer? The idea would be to open the link in the previously selected buffer, to keep the org-roam buffer open.

**Comment by jethrokuan** [comment]
Yeah, but actually I'm thinking of eventually moving away from using org-mode as the major mode for the org-roam buffer, to magit-section where things can be customized more easily, so this problem should go away. See #5

**Comment by l3kn** [comment]
A workaround that seems to work well is storing the window `org-roam` was called from in a variable,
then use a custom link type in the org-roam buffer

``` lisp
(defun org-roam-find-file (file)
  (with-selected-window org-roam-last-window
    (find-file file))
  (select-window org-roam-last-window))

(org-link-set-parameters "org_roam_file" :follow #'org-roam-find-file)
```

@jethrokuan Does that seem like a good solution to you? If so, I'll open a pull request for it.

**Comment by jethrokuan** [comment]
@l3kn that does look like a good solution. Happy to have a PR. Thanks!

**Comment by jethrokuan** [comment]
This is closed with #108 thanks @l3kn
